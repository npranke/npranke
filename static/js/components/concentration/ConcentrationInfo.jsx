import { getSendEventHandler } from '@utils'

function ConcentrationInfo() {
    return (
        <div className="concetration-info">
            <p>
                Concentration is a picture matching memory game.
            </p>
            <p>
                Click on the tiles to reveal matching pictures. Find all 12
                matches as quickly as you can.
            </p>
            <p>
                Concentration is built with images from NASA&#8217;s Mars
                Exploration Program. The picture on the back of each tile
                was taken with the hazard avoidance camera on the Mars Rover
                Opportunity. The matching pictures on the tiles were taken
                with various instruments on the Mars Reconnaissance Orbiter.
                Learn more about Opportunity, the MRO, and other Mars missions
                by visiting&nbsp;
                <a
                    onClick={ getSendEventHandler(
                        'concentration',
                        'navigate',
                        'NASA Mars Exploration Program',
                    ) }
                    href="https://mars.nasa.gov"
                    target="_blank"
                    rel="noreferrer noopener external"
                >
                    NASA&#8217;s Mars Exploration Program
                </a>.
            </p>
            <p>
                The images used in Concentration are courtesy of
                NASA/JPL-Caltech. Discover more images by visiting
                the&nbsp;
                <a
                    onClick={ getSendEventHandler(
                        'concentration',
                        'navigate',
                        'JPL Image Gallery',
                    ) }
                    href="https://www.jpl.nasa.gov/images"
                    target="_blank"
                    rel="noreferrer noopener external"
                >
                    JPL Image Gallery
                </a>.
            </p>
        </div>
    )
}

export default ConcentrationInfo
