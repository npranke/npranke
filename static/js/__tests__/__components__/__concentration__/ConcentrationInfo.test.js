import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'
import { render } from '@testing-library/react'

import ConcentrationInfo from '@components/concentration/ConcentrationInfo'

Enzyme.configure({ adapter: new Adapter() })

describe('ConcentrationInfo', () => {
    test('has basic description text', () => {
        const concentrationInfo = shallow(<ConcentrationInfo />)

        expect(
            concentrationInfo.find('.concetration-info').text(),
        ).toContain('picture matching memory game')
    })

    test('has image credit text', () => {
        const concentrationInfo = shallow(<ConcentrationInfo />)

        expect(
            concentrationInfo.find('.concetration-info').text(),
        ).toContain('courtesy of NASA/JPL-Caltech')
    })

    test('has link to nasa mars exploration program', () => {
        const concentrationInfo = shallow(<ConcentrationInfo />)

        expect(
            concentrationInfo.find(
                '.concetration-info a',
            ).first().props().href,
        ).toEqual('https://mars.nasa.gov')
    })

    test('has link to jpl image gallery', () => {
        const concentrationInfo = shallow(<ConcentrationInfo />)

        expect(
            concentrationInfo.find(
                '.concetration-info a',
            ).last().props().href,
        ).toEqual('https://www.jpl.nasa.gov/images')
    })
})

describe('ConcentrationInfo snapshot', () => {
    test('matches snapshot', () => {
        const { asFragment } = render(
            <ConcentrationInfo />,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
