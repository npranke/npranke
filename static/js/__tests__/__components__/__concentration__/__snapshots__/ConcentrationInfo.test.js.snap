// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcentrationInfo snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="concetration-info"
  >
    <p>
      Concentration is a picture matching memory game.
    </p>
    <p>
      Click on the tiles to reveal matching pictures. Find all 12 matches as quickly as you can.
    </p>
    <p>
      Concentration is built with images from NASA’s Mars Exploration Program. The picture on the back of each tile was taken with the hazard avoidance camera on the Mars Rover Opportunity. The matching pictures on the tiles were taken with various instruments on the Mars Reconnaissance Orbiter. Learn more about Opportunity, the MRO, and other Mars missions by visiting 
      <a
        href="https://mars.nasa.gov"
        rel="noreferrer noopener external"
        target="_blank"
      >
        NASA’s Mars Exploration Program
      </a>
      .
    </p>
    <p>
      The images used in Concentration are courtesy of NASA/JPL-Caltech. Discover more images by visiting the 
      <a
        href="https://www.jpl.nasa.gov/images"
        rel="noreferrer noopener external"
        target="_blank"
      >
        JPL Image Gallery
      </a>
      .
    </p>
  </div>
</DocumentFragment>
`;
